### PR TITLE
Bump to 2.29.0

### PR DESCRIPTION
**Pull request by matthewfala**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: CHANGELOG.md
Comment (PettitWesley) [inline CHANGELOG.md:13]: this hsould be called out as a `Bug` entry

File: CHANGELOG.md
Comment (PettitWesley) [inline CHANGELOG.md:19]: nit: `Use log timestamp for s3 key timestamp` is more clear IMO

**Review by PettitWesley — DISMISSED** [review]

File: CHANGELOG.md
Comment (matthewfala) [reply CHANGELOG.md:19]: Great. Thanks

**Review by PettitWesley — CHANGES_REQUESTED** [review]
Blocking this merge until @claych PRs are approved and tested. If we want to still release, we need to remove those cherry picks.

**Review by PettitWesley — APPROVED** [review]
@matthewfala Good idea 💪, splitting up many changes into two releases; I should've thought of that myself.

**Pull request merged**
